arrays/arrays.c

```diff
@@ -83,15 +107,23 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
+  if (index > arr->count)
+  {
+    fprintf(stderr, "IndexError: Index %d is out of range\n", index);
+    exit(1);
+  }
 
   // Resize the array if the number of elements is over capacity
+  if (arr->count == (arr->capacity))
```

Comment:
Generally speaking this shouldn't cause an issue, but because of the public nature of these variables, in my mind I can imagine someone could set arr->count to a number greater than arr->capacity before calling this function. Which means this condition would never be hit and the resizing would not take place as needed to prevent an segmentation fault or similar issue with accessing non-allocated memory. In other words I think it'd be best to use `arr->count >= arr->capacity` instead of the strictly equal to condition.